### PR TITLE
Collegiate subreddit get post

### PR DESCRIPTION
**Pull request by lukeyoffe**
https://github.com/ucsb-cs156-w22/team02-w22-5pm-1/issues/6

### CODE REVIEW

File: src/main/java/edu/ucsb/cs156/team02/controllers/CollegiateSubredditController.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    @GetMapping("all")
+    public Iterable<CollegiateSubreddit> allCollegiateSubreddits(){
+        loggingService.logMethod();
+        Iterable<CollegiateSubreddit> subreddits = collegiateSubredditRepository.findAll();
+        return subreddits;
+    }
+
+    @ApiOperation(value = "Creates a new subreddit")
+    @PreAuthorize("hasRole('ROLE_USER')")
+    @PostMapping("post")
```
Comment (sidd328) [inline src/main/java/edu/ucsb/cs156/team02/controllers/CollegiateSubredditController.java]: believe this should be `/post`

File: src/test/java/edu/ucsb/cs156/team02/controllers/CollegiateSubredditControllerTests.java [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+@Import(TestConfig.class)
+public class CollegiateSubredditControllerTests extends ControllerTestCase {
+    
+    @MockBean
+    CollegiateSubredditRepository collegiateSubredditRepository;
+
+    @MockBean
+    UserRepository userRepository;
+
+    @WithMockUser(roles = { "USER" })
```
Comment (sidd328) [inline src/test/java/edu/ucsb/cs156/team02/controllers/CollegiateSubredditControllerTests.java:40]: Can you also do a test for each type for admin?

Comment (lukeyoffe) [reply src/test/java/edu/ucsb/cs156/team02/controllers/CollegiateSubredditControllerTests.java:40]: I talked to Andrew and he gave me the impression that for this project, we don't have to do anything with admins

File: src/main/java/edu/ucsb/cs156/team02/controllers/CollegiateSubredditController.java (hunk 1 above)
Comment (lukeyoffe) [reply src/main/java/edu/ucsb/cs156/team02/controllers/CollegiateSubredditController.java]: Good idea, I added the "/".  It seems to work without the slash, but may as well have it to be consistent.

**Review by sidd328 — APPROVED** [review]
LGTM

**Pull request merged**